text/0000-stack-metadata.md

```diff
@@ -0,0 +1,138 @@
+```
+Inspecting image: go-mod-simple-app
+...................
+LOCAL:
+
+Stack: io.buildpacks.stacks.bionic
+
+Base Image:
+  Reference: 610caf01e85e5da52c6a99b42fed80858c153de6d25b120940266846bec0872a
+  Top Layer: sha256:b6aadae5b30af0345f0ab82add7ed5b951d78230e86693c08f5c1ae8470a0ffb
+
+Run Images:
+  gcr.io/paketo-buildpacks/run:base-cnb
+...................
+```
+
+By adding the proposed metadata, `pack inspect-image` output of this app could eventually look like:
+```
+Inspecting image: go-mod-simple-app
+...................
+LOCAL:
+
+Stack: io.buildpacks.stacks.bionic
+Stack Metadata: 
+  Maintainer: "Paketo Buildpacks"
+  Homepage: "https://github.com/paketo-buildpacks/stacks"
+  OS Distro: "unbuntu 18.04"
+
+Base Image:
+  Reference: 610caf01e85e5da52c6a99b42fed80858c153de6d25b120940266846bec0872a
+  Top Layer: sha256:b6aadae5b30af0345f0ab82add7ed5b951d78230e86693c08f5c1ae8470a0ffb
+
+Run Images:
+  gcr.io/paketo-buildpacks/run:base-cnb
+...................
+```
+
+# What it is
+[what-it-is]: #what-it-is
+
+This provides a high level overview of the feature.  
+
+- Define the target persona: stack author  
+- Explaining the feature largely in terms of examples.  
+The following labels can **optionally** be added by stack maintainers to stack images:  
+`io.buildpacks.stack.maintainer`: Name of Stack Maintainer  
+`io.buildpacks.stack.homepage`: URL for the stack (i.e repo link)  
+`io.buildpacks.stack.distro.name`: Name of OS Distribution  
+`io.buildpacks.stack.distro.version`: Version of OS Distribution  
+`io.buildpacks.stack.version`: Release Number  
+`io.buildpacks.stack.release_date`: Release date of image  
+`io.buildpacks.stack.description`: Description  
+`io.buildpacks.stack.metadata`: Generic stack metadata
+
+
+# How it Works  
+[how-it-works]: #how-it-works  
+
+This is the technical portion of the RFC, where you explain the design in sufficient detail.  
+All labels listed above will be added as `Labels` in the stack's Dockerfile.   
+
+Example Dockerfile:
+```
+ARG cnb_uid=1000
+ARG cnb_gid=1000
+ARG stack_id="io.buildpacks.stacks.bionic"
+ENV CNB_USER_ID=${cnb_uid}
+ENV CNB_GROUP_ID=${cnb_gid}
+ENV CNB_STACK_ID=${stack_id}
+USER ${cnb_uid}:${cnb_gid}
+LABEL io.buildpacks.stack.id=${stack_id}
+LABEL io.buildpacks.stack.mixins="[\"ca-certificates\"............]"
+LABEL io.buildpacks.stack.maintainer="Paketo Buildpacks"
+LABEL io.buildpacks.stack.homepage="https://github.com/paketo-buildpacks/stacks/"
+LABEL io.buildpacks.stack.distro.name="ubuntu"
+LABEL io.buildpacks.stack.distro.version="18.04"
+LABEL io.buildpacks.stack.version="25"
+LABEL io.buildpacks.stack.release_date="2020-05-12T05:17:02.390472"
+LABEL io.buildpacks.stack.description="Paketo Buildpacks base stack build image"
+LABEL io.buildpacks.stack.metadata="Some optionally defined metadata"
+```
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+This could make image rebasing slightly more complicated as labels like stack version will need to be updated.  
```

Comment:
How would rebase work with this RFC?

Reply:
I believe the `io.buildpacks.stack.*` labels would need to be updated on the app image.

Reply:
That sounds good to me. Can get how this works into the RFC itself?